scripts/languageserver/provider_misc.jl

```diff
@@ -75,10 +75,36 @@ end
 abstract didChange <: Method
 
 function Respond(r::Request{didChange,DidChangeTextDocumentParams})
+        vtdi = r.params.textDocument
+        C = r.params.contentChanges
+        for c in C
+            c(vtdi)
+        end
     try
-        documents[r.params.textDocument.uri] = split(r.params.contentChanges[1].text, r"\r\n?|\n")
         return
     catch err
         return Response{didChange,Exception}("2.0",r.id,err)
     end
 end
+
+function (c::TextDocumentContentChangeEvent)(vtdi::VersionedTextDocumentIdentifier)
+    d = join(documents[vtdi.uri],'\n').data
```

Comment:
Ah, you are right! Well, but what we could do is simply start out with an array of ``SubString``s, and then as we receive change notifications, we could replace those ``SubString``s with real ``String``s for those lines that get modified?